src/glob.zig

```diff
@@ -0,0 +1,3021 @@
+        error_on_broken_symlinks: bool,
+        only_files: bool,
+    ) !Maybe(void) {
+        errdefer arena.deinit();
+        var cwd: BunString = undefined;
+        switch (Syscall.getcwd(&this.pathBuf)) {
+            .err => |err| {
+                return .{ .err = err };
+            },
+            .result => |result| {
+                var copiedCwd = try arena.allocator().alloc(u8, result.len);
+                @memcpy(copiedCwd, result);
+                cwd = BunString.fromBytes(copiedCwd);
+            },
+        }
+        const globWalker = try this.initWithCwd(
+            arena,
+            pattern,
+            cwd,
+            dot,
+            absolute,
+            follow_symlinks,
+            error_on_broken_symlinks,
+            only_files,
+        );
+        return .{ .result = globWalker };
+    }
+
+    pub fn convertUtf8ToCodepoints(codepoints: []u32, pattern: []const u8) void {
+        switch (comptime @import("builtin").target.cpu.arch.endian()) {
+            .Big => {
+                _ = bun.simdutf.convert.utf8.to.utf32.be(pattern, codepoints);
+            },
+            .Little => {
+                _ = bun.simdutf.convert.utf8.to.utf32.le(pattern, codepoints);
+            },
+        }
+    }
+
+    /// `cwd` should be allocated with the arena
+    pub fn initWithCwd(
+        this: *GlobWalker,
+        arena: *Arena,
+        pattern: []const u8,
+        cwd: BunString,
+        dot: bool,
+        absolute: bool,
+        follow_symlinks: bool,
+        error_on_broken_symlinks: bool,
+        only_files: bool,
+    ) !void {
+        var patternComponents = ArrayList(Component){};
+        try GlobWalker.buildPatternComponents(
+            arena,
+            &patternComponents,
+            pattern,
+            &this.cp_len,
+            &this.pattern_codepoints,
+            &this.has_relative_components,
+        );
+
+        this.patternComponents = patternComponents;
+        this.pattern = pattern;
+        this.arena = arena.*;
+        this.cwd = cwd;
+        this.dot = dot;
+        this.absolute = absolute;
+        this.follow_symlinks = follow_symlinks;
+        this.error_on_broken_symlinks = error_on_broken_symlinks;
+        this.only_files = only_files;
+    }
+
+    pub fn deinit(this: *GlobWalker) void {
+        this.arena.deinit();
+    }
+
+    pub fn handleSysErrWithPath(
+        this: *GlobWalker,
+        err: Syscall.Error,
+        path_buf: [:0]const u8,
+    ) Syscall.Error {
+        @memcpy(this.pathBuf[0 .. path_buf.len + 1], @as([]const u8, @ptrCast(path_buf[0 .. path_buf.len + 1])));
+        return err.withPath(this.pathBuf[0 .. path_buf.len + 1]);
+    }
+
+    pub fn walk(this: *GlobWalker) !Maybe(u0) {
+        if (this.patternComponents.items.len == 0) return .{ .result = 0 };
+        var path_buf: [bun.MAX_PATH_BYTES]u8 = std.mem.zeroes([bun.MAX_PATH_BYTES]u8);
+
+        const root_path = this.cwd.toZigString().slice();
```

Comment:
this is going to leak memory or not be thread safe (must call `.toThreadSafe` before `BunString` can be accessed on another thread)

you should copy the directory as a string IMO